server/src/main/java/org/opensearch/index/codec/CodecService.java

```diff
@@ -58,15 +60,21 @@ public class CodecService {
     public static final String BEST_COMPRESSION_CODEC = "best_compression";
     /** the raw unfiltered lucene default. useful for testing */
     public static final String LUCENE_DEFAULT_CODEC = "lucene_default";
+    public static final String ZSTD_CODEC = "zstd";
+    public static final String ZSTD_NO_DICT_CODEC = "zstd_no_dict";
 
     public CodecService(@Nullable MapperService mapperService, Logger logger) {
         final MapBuilder<String, Codec> codecs = MapBuilder.<String, Codec>newMapBuilder();
         if (mapperService == null) {
             codecs.put(DEFAULT_CODEC, new Lucene95Codec());
             codecs.put(BEST_COMPRESSION_CODEC, new Lucene95Codec(Mode.BEST_COMPRESSION));
+            codecs.put(ZSTD_CODEC, new ZstdCodec());
+            codecs.put(ZSTD_NO_DICT_CODEC, new ZstdNoDictCodec());
         } else {
             codecs.put(DEFAULT_CODEC, new PerFieldMappingPostingFormatCodec(Mode.BEST_SPEED, mapperService, logger));
             codecs.put(BEST_COMPRESSION_CODEC, new PerFieldMappingPostingFormatCodec(Mode.BEST_COMPRESSION, mapperService, logger));
+            codecs.put(ZSTD_CODEC, new ZstdCodec(mapperService, logger));
+            codecs.put(ZSTD_NO_DICT_CODEC, new ZstdNoDictCodec(mapperService, logger));
         }
         codecs.put(LUCENE_DEFAULT_CODEC, Codec.getDefault());
         for (String codec : Codec.availableCodecs()) {
```

Comment:
Question here, `custom-codecs` went into 2.8 release (as sandbox plugin) and potentially could be installed on some clusters. Now we move it to core but the plugin still could be installed, should we do a safety check and verify there is no conflict between available codecs and preregistered ones?

Reply:
I think we should be good on this front since we don't provide any guarantees with the sandbox features as mentioned in #570

```
Aside from ensured security (as a by product of the release process) this module will not have any other guarantees; no backwards compatibility, no long term support, can be removed at any time. But it does give a nice place to keep innovation moving forward.
```